### ISSUE TITLE
CollectionView.emptyView not shown when filtered collection is empty

### ISSUE THREAD

**Issue by jackocnr** [marionettejs/backbone.marionette#2437]
When **not** using CollectionView.filter, if you remove the last model from the collection, the CollectionView will automatically display the emptyView. When you **are** using CollectionView.filter however, if you remove the last model that matches the filter, it doesn't display the emptyView, presumably because the pre-filtered collection is not empty. This leaves you with a broken empty state (no childViews are rendered, but no emptyView is rendered either).

Demo here: http://codepen.io/jackocnr/pen/wBRyrE?editors=101

Workaround:
```js
onRemoveChild: function() {
if (!this.children.length) { this.showEmptyView(); }
}
```

**Comment by cmaher** [comment]
It looks like `isEmpty` needs to be updated to account for the filter: https://github.com/marionettejs/backbone.marionette/blob/master/src/collection-view.js#L462

```js
if (!this.collection) { return true; }
var models = this.collection.filter(function(model, index) {
        return this._shouldAddChild(model, index);
      }, this);
return models.length === 0;
```

**Comment by cmaher** [comment]
resolved by #2437
